torch/csrc/jit/passes/quantization.cpp

```diff
@@ -572,5 +601,80 @@ template TORCH_API void InsertQuantDequantNodesForParam(
         getQParamFunc,
     at::ScalarType t);
 
+
+TORCH_API script::Module InsertObservers(
+    const script::Module& module,
+    const std::string& method_name,
+    const script::Module& observer_module,
+    const script::Module& weight_observer_module) {
+  script::Module input_module = module.clone();
+  script::Method method = input_module.get_method(method_name);
+  auto graph = method.graph();
+  TORCH_CHECK(graph != nullptr);
+  // For storing all values that need to be instrumented with an observer call.
+  std::vector<Value*> values_to_observe;
+
+  // For traversing all blocks in the graph including subblocks.
+  std::stack<Block*> blocks_to_visit;
+
+  // Mark observer nodes for inputs so we dont add observers
+  // for observers while traversing graph
+  std::unordered_set<Node*> observer_for_input;
+
+  // Add observer for external input nodes excluding parameters
+  // These are treated as activation as they vary across batches
+  // and need to be observed.
+
+  // prim::Param nodes do not belong to the graph. Hence the Insert
+  // point is the beginning of graph node. This also safe guards against
+  // observing a potentially mutated value due to some in-place operation
+  for (size_t idx = 1; idx < method.num_inputs(); ++idx) {
+    auto& v = graph->inputs()[idx];
```

Comment:
Are we always inserting observers at inputs? This should be optional, with default = True.

Reply:
we can consider corner cases later, this is a primitive version that makes many assumptions